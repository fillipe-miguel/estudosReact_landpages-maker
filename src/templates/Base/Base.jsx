import P from 'prop-types';
import React from 'react';

import { GoTop } from '../../components';
import { LayoutFooter, LayoutHeader } from '../../layouts';

import * as Styled from './styles';

function Base({ links = [], logoData, footerHtml, children }) {
  return (
    <>
      <Styled.BaseRoot>
        <LayoutHeader links={links} logoData={logoData} />
        {children}
        <LayoutFooter html={footerHtml} />
      </Styled.BaseRoot>
      <GoTop />
    </>
  );
}

Base.defaultProps = {

};

Base.propTypes = {
  ...LayoutHeader.propTypes,
  children: P.node.isRequired,
  footerHtml: P.string.isRequired,
};

export default Base;
