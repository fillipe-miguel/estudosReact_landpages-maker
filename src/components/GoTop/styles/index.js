import styled, { css } from 'styled-components';

export const GoTopRoot = styled.a`
  ${({ theme }) => css`
    align-items: center;
    background-color: ${theme.colors.primaryColor};
    border-radius: 1rem;
    border: 2px solid ${theme.colors.white};
    bottom: 2rem;
    color: ${theme.colors.white};
    display: flex;
    font-size: ${theme.fonts.sizes.medium};
    height: 4rem;
    justify-content: center;
    position: fixed;
    right: 2rem;
    opacity: 0.5;
    z-index: 6;
    transition: all 0.3s ease-in-out;
    width: 4rem;

    &:hover {
      opacity: 1;
      border-color: ${theme.colors.secondaryColor};
      box-shadow: 0px 0px 8px 1px ${theme.colors.secondaryColor};
      color: ${theme.colors.secondaryColor};
      transform: scale(1.05);
      transition: all 0.3s ease-in-out;
    }

    &:hover > :first-child {
      animation: shakeUpDown 0.7s ease-in-out infinite;
    }
  `}

  @keyframes shakeUpDown {
    from {
      transform: translateY(0px);
    }
    to {
      transform: translateY(-5px);
    }
  }
`;

export default GoTopRoot;
