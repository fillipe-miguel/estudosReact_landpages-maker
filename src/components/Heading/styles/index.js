import styled, { css } from 'styled-components';

export const HeadingRoot = styled.h1`
  // ================================= Para pegar as props =================================
  ${({ theme, ...props }) => css`
    color: ${props.colorDarker
      ? theme.colors.primaryColor
      : theme.colors.white};

    text-transform: ${props.uppercase ? 'uppercase' : 'none'};

    font-size: ${theme.fonts.sizes[props.size]};
  `}

  // ================================= Medias =================================
  ${({ theme, ...props }) => css`
    @media ${theme.medias.lteMedium} {
      font-size: ${props.size === 'xxxlarge'
        ? theme.fonts.sizes.xlarge
        : theme.fonts.sizes[props.size]};
    }
  `}
`;

export default HeadingRoot;

// ================================= Antiga organização =================================

// font-size: ${({ theme, size }) => titleSize[size](theme)};

// @media ${({ theme }) => theme.media.lteMedium} {
//   font-size: ${({ theme, size }) =>
//     size === 'xxxlarge'
//       ? theme.fonts.sizes.xlarge
//       : theme.fonts.sizes.xxxlarge};
// }

// color: ${({ theme, colorDarker }) =>
//     colorDarker ? theme.colors.black : theme.colors.white};

// text-transform: ${({ uppercase }) => (uppercase ? 'uppercase' : 'none')};
// font-size: ${({ theme, size }) => theme.fonts.sizes[size]};
