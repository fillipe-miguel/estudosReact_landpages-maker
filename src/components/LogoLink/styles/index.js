import styled, { css } from 'styled-components';

export const LogoLinkRoot = styled.a`
  ${() => css`
    display: flex;
    align-items: center;

    > img {
      max-height: 3rem;
    }
  `}
`;

export default LogoLinkRoot;
