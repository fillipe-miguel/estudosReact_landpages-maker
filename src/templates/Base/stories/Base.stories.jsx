import Base from '..';
import {
  LayoutCenterContent,
  LayoutGridColumn,
  LayoutGridGallery,
} from '../../../layouts';

import * as mocks from '../../../mocks';

export const baseArgs = {
  links: mocks.components.links,
  logoData: mocks.components.logoData,
  footerHtml: '<p>Bah que loucura</p>',
  children: (
    <>
      <LayoutCenterContent
        {...mocks.layouts.layoutCenterContent}
        setBackground
      />
      <LayoutGridGallery {...mocks.layouts.gridGallery} />
      <LayoutGridColumn {...mocks.layouts.layoutGridColumn} setBackground />
      <LayoutGridGallery {...mocks.layouts.gridGallery} />
    </>
  ),
}

export default {
  title: 'Templates/Base',
  component: Base,
  args: baseArgs,
};

export function Render01(args) {
  return <Base {...args} />;
}
