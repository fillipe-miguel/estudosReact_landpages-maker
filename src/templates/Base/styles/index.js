import styled, { css } from 'styled-components';

export const BaseRoot = styled.div`
  ${({ theme }) => css`
    /* padding-top: 9rem; */

    @media ${theme.medias.lteMedium} {
      padding-top: 0;
    }
  `}
`;

export default BaseRoot;
