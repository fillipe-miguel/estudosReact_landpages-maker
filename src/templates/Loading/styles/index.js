import styled, { css } from 'styled-components';

export const LoadingRoot = styled.div`
  ${() => css`
    display: flex;
    flex-direction: column;
    gap: 2rem;
    align-items: center;
    justify-content: center;
  `}
`;

export default LoadingRoot;
