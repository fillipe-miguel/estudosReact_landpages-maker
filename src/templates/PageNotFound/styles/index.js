import styled, { css } from 'styled-components';
import { TextComponentStyles } from '../../../components';

export const PageNotFoundRoot = styled.div`
  ${({ theme }) => css`
    & ${TextComponentStyles.TextComponentRoot} {
      font-size: ${theme.fonts.sizes.large};
      text-align: center;
    }
  `}
`;

export default PageNotFoundRoot;
