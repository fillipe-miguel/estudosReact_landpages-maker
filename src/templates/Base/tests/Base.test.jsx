/* eslint-disable no-undef */
import { render, screen } from '@testing-library/react';

import { MyThemeProvider } from '../../../styles';

import Base from '../Base';

// Um mock criado para aglomerar as coisas
import { baseArgs } from '../stories/Base.stories';

describe('<Base />', () => {
  // ================================= Teste 01 =================================
  it('Should match snapshot', () => {
    const { container } = render(
      <MyThemeProvider>
        <Base {...baseArgs} />
      </MyThemeProvider>,
    );

    expect(container).toMatchSnapshot();
  });

  // ================================= Teste 02 =================================
  it('Should render whit links modifier', () => {
    render(
      <MyThemeProvider>
        <Base
          {...baseArgs}
          links={undefined}
          logoData={{
            text: 'Teste',
            link: '#',
            srcImage: '',
          }}
        />
      </MyThemeProvider>,
    );

    expect(screen.queryByRole('heading', { name: 'Teste' })).toBeInTheDocument();
  });
});
