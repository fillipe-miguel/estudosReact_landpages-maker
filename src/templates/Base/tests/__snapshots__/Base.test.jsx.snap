// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Base /> Should match snapshot 1`] = `
.c28 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #0A1128;
  border-radius: 1rem;
  border: 2px solid #FFF;
  bottom: 2rem;
  color: #FFF;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 2.4rem;
  height: 4rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: fixed;
  right: 2rem;
  opacity: 0.5;
  z-index: 6;
  -webkit-transition: all 0.3s ease-in-out;
  transition: all 0.3s ease-in-out;
  width: 4rem;
}

.c28:hover {
  opacity: 1;
  border-color: #DC143C;
  box-shadow: 0px 0px 8px 1px #DC143C;
  color: #DC143C;
  -webkit-transform: scale(1.05);
  -ms-transform: scale(1.05);
  transform: scale(1.05);
  -webkit-transition: all 0.3s ease-in-out;
  transition: all 0.3s ease-in-out;
}

.c28:hover > :first-child {
  -webkit-animation: shakeUpDown 0.7s ease-in-out infinite;
  animation: shakeUpDown 0.7s ease-in-out infinite;
}

.c8 {
  color: #0A1128;
  text-transform: uppercase;
  margin-bottom: 3.2rem;
  font-size: 2.4rem;
}

.c13 {
  color: #FFF;
  text-transform: uppercase;
  margin-bottom: 3.2rem;
  font-size: 6.4rem;
}

.c19 {
  color: #0A1128;
  text-transform: uppercase;
  margin-bottom: 3.2rem;
  font-size: 4.8rem;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  max-height: 8rem;
  width: 20rem;
}

.c10 {
  display: inline-block;
  -webkit-text-decoration: none;
  text-decoration: none;
  position: relative;
  font-size: 1.6rem;
  padding: 1.6rem;
  color: #0A1128;
}

.c10::after {
  content: '';
  position: absolute;
  bottom: 1rem;
  left: 50%;
  width: 0;
  height: 0.2rem;
  background-color: #DC143C;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c10:hover::after {
  left: 25%;
  width: 50%;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
}

.c16 {
  font-size: 1.6rem;
  color: #FFF;
}

.c16 p {
  margin: 2.4rem 0;
}

.c16 a {
  font-weight: bolder;
  color: #DC143C;
}

.c16 a:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c16 h1,
.c16 h2,
.c16 h3,
.c16 h4,
.c16 h5,
.c16 h6 {
  margin: 1.6rem 0;
}

.c16 ol,
.c16 ul {
  margin: 1.6rem;
}

.c16 ol > li,
.c16 ul > li {
  margin: 1.6rem 0;
  font-style: italic;
}

.c20 {
  font-size: 1.6rem;
  color: #0A1128;
}

.c20 p {
  margin: 2.4rem 0;
}

.c20 a {
  font-weight: bolder;
  color: #DC143C;
}

.c20 a:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c20 h1,
.c20 h2,
.c20 h3,
.c20 h4,
.c20 h5,
.c20 h6 {
  margin: 1.6rem 0;
}

.c20 ol,
.c20 ul {
  margin: 1.6rem;
}

.c20 ol > li,
.c20 ul > li {
  margin: 1.6rem 0;
  font-style: italic;
}

.c12 {
  text-align: center;
  max-width: 58rem;
  margin: 0 auto;
}

.c14 {
  margin-top: 3.2rem;
}

.c14 h1,
.c14 h2,
.c14 h3,
.c14 h4,
.c14 h5,
.c14 h6 {
  text-align: center;
}

.c4 {
  max-width: 120rem;
  margin: 0 auto;
  width: 100%;
  padding: 3.2rem;
}

.c11 {
  min-height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #FFF;
  color: #0A1128;
  background-color: #0A1128;
  color: #FFF;
}

.c17 {
  min-height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #FFF;
  color: #0A1128;
}

.c27 {
  min-height: 8rem;
  text-align: center;
  background-color: #FFF;
  border-top: 1px solid #DDDDDD;
}

.c27 .c15 {
  font-size: 1.6rem;
}

.c27 .c15 p {
  margin: 0;
}

.c27 .c3 {
  padding-top: 2rem;
  padding-bottom: 2rem;
}

.c23 {
  display: grid;
  grid-template-columns: 1fr 2fr;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 4.8rem;
}

.c23 .c7 {
  margin-bottom: 1.6rem;
}

.c24 {
  min-width: 35rem;
}

.c25 {
  max-width: 74rem;
}

.c26 {
  max-width: 100%;
}

.c18 {
  margin-bottom: 4.8rem;
}

.c21 {
  display: grid;
  grid-template-columns: repeat(auto-fill,minmax(270px,1fr));
  gap: 2.4rem;
}

.c22 {
  overflow: hidden;
  height: -webkit-fit-content;
  height: -moz-fit-content;
  height: fit-content;
}

.c22:hover > img {
  -webkit-transition: all 0.3s ease-in-out;
  transition: all 0.3s ease-in-out;
  -webkit-transform: scale(1.2) rotate(9deg);
  -ms-transform: scale(1.2) rotate(9deg);
  transform: scale(1.2) rotate(9deg);
}

.c22 img {
  pointer-events: none;
  max-width: 100%;
}

.c2 {
  position: fixed;
  z-index: 5;
  top: 0;
  right: 0;
  left: 0;
  background-color: #FFF;
  border-bottom: 1px solid #DDDDDD;
}

.c2 > .c3 {
  padding-top: 1.6rem;
  padding-bottom: 1.6rem;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c1 {
  z-index: 6;
  position: fixed;
  top: 2rem;
  right: 2rem;
  display: none;
  font-size: 2.5rem;
  padding: 0.5rem;
  cursor: pointer;
  border: none;
  background-color: #0A1128;
  color: #FFF;
}

@media (max-width:768px) {
  .c8 {
    font-size: 2.4rem;
  }
}

@media (max-width:768px) {
  .c13 {
    font-size: 4rem;
  }
}

@media (max-width:768px) {
  .c19 {
    font-size: 4.8rem;
  }
}

@media (max-width:768px) {
  .c9 {
    -webkit-flex-flow: column nowrap;
    -ms-flex-flow: column nowrap;
    flex-flow: column nowrap;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
  }
}

@media (max-width:768px) {
  .c4 {
    padding-inline: 3.2rem;
  }
}

@media (max-width:768px) {
  .c23 {
    grid-template-columns: 1fr;
  }

  .c23 .c7 {
    margin-bottom: 4rem;
  }
}

@media (max-width:768px) {
  .c24 {
    min-width: auto;
    text-align: center;
    margin-bottom: 3.2rem;
  }
}

@media (max-width:768px) {
  .c18 {
    text-align: center;
  }
}

@media (max-width:768px) {
  .c21 {
    grid-template-columns: 1fr;
  }
}

@media (max-width:768px) {
  .c2 {
    bottom: 0;
    height: 100vh;
    visibility: hidden;
    opacity: 0;
    -webkit-transition: all 0.3s ease-in-out;
    transition: all 0.3s ease-in-out;
  }

  .c2 > .c3 {
    display: grid;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    height: 100%;
    margin: 0;
    overflow-y: auto;
  }

  .c2 .c7 {
    padding-bottom: 3.2rem;
  }
}

@media (max-width:768px) {
  .c5 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    gap: 3rem;
    text-align: center;
    padding: 4.8rem;
  }
}

@media (max-width:768px) {
  .c1 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
  }
}

@media (max-width:768px) {
  .c0 {
    padding-top: 0;
  }
}

<div>
  <div
    class="c0"
  >
    <button
      aria-label="Open/Close Menu"
      class="c1"
      title="Open menu"
    >
      <svg
        aria-label="Open menu icon"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0V0z"
          fill="none"
        />
        <path
          d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
        />
      </svg>
    </button>
    <header
      class="c2"
    >
      <div
        class="c3 c4"
      >
        <div
          class="c5"
        >
          <a
            class="c6"
            href="#target"
          >
            <h2
              class="c7 c8"
            >
              logo link
            </h2>
          </a>
          <nav
            aria-label="Main menu"
            class="c9"
          >
            <a
              class="c10"
              href="https://www.google.com.br"
              target="_blank"
            >
              Google
            </a>
            <a
              class="c10"
              href="https://youtube.com"
              target="_blank"
            >
              YouTube
            </a>
            <a
              class="c10"
              href="https://github.com/fillipe-miguel"
              target="_self"
            >
              My GitHub
            </a>
            <a
              class="c10"
              href="https://www.tiktok.com/pt-BR/"
              target="_blank"
            >
              TikTok
            </a>
            <a
              class="c10"
              href="https://www.udemy.com/"
              target="_blank"
            >
              Udemy
            </a>
          </nav>
        </div>
      </div>
    </header>
    <div
      class="c11"
      id="teste"
    >
      <div
        class="c3 c4"
      >
        <div
          class="c12"
        >
          <h2
            class="c7 c13"
          >
            Layout Center
          </h2>
          <div
            class="c14"
          >
            <div
              class="c15 c16"
            >
              
  
              <p>
                <strong>
                  Pellentesque habitant morbi tristique
                </strong>
                 senectus et netus et malesuada fames ac turpis egestas. Vestibulum tortor quam, feugiat vitae, ultricies eget, tempor sit amet, ante. Donec eu libero sit amet quam egestas semper. 
                <em>
                  Aenean ultricies mi vitae est.
                </em>
                 Mauris placerat eleifend leo. Quisque sit amet est et sapien ullamcorper pharetra. Vestibulum erat wisi, condimentum sed, 
                <code>
                  commodo vitae
                </code>
                , ornare sit amet, wisi. Aenean fermentum, elit eget tincidunt condimentum, eros ipsum rutrum orci, sagittis tempus lacus enim ac dui. 
                <a
                  href="#"
                >
                  Donec non enim
                </a>
                 in turpis pulvinar facilisis. Ut felis.
              </p>
              

  
              <h2>
                Header Level 2
              </h2>
              

  
              <ol>
                
     
                <li>
                  Lorem ipsum dolor sit amet, consectetuer adipiscing elit.
                </li>
                
     
                <li>
                  Aliquam tincidunt mauris eu risus.
                </li>
                
  
              </ol>
              

  
              <blockquote>
                <p>
                  Lorem ipsum dolor sit amet, consectetur adipiscing elit. Vivamus magna. Cras in mi at felis aliquet congue. Ut a est eget ligula molestie gravida. Curabitur massa. Donec eleifend, libero at sagittis mollis, tellus est malesuada tellus, at luctus turpis elit sit amet quam. Vivamus pretium ornare est.
                </p>
              </blockquote>
              

  
              <h3>
                Header Level 3
              </h3>
              

  
              <ul>
                
     
                <li>
                  Lorem ipsum dolor sit amet, consectetuer adipiscing elit.
                </li>
                
     
                <li>
                  Aliquam tincidunt mauris eu risus.
                </li>
                
  
              </ul>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="c17"
      id="teste"
    >
      <div
        class="c3 c4"
      >
        <div
          class="c18"
        >
          <h2
            class="c7 c19"
          >
            My grid gallery
          </h2>
          <p
            class="c15 c20"
          >
            Aqui vai ter várias imagens e coisas legais e esse texto aqui é para testar a parada da responsividade vem comigo neném
          </p>
        </div>
        <div
          class="c21"
        >
          <div
            class="c22"
          >
            <img
              alt="imagem 1"
              src="https://source.unsplash.com/random/800x800?r=1"
            />
          </div>
          <div
            class="c22"
          >
            <img
              alt="imagem 2"
              src="https://source.unsplash.com/random/800x800?r=2"
            />
          </div>
          <div
            class="c22"
          >
            <img
              alt="imagem 3"
              src="https://source.unsplash.com/random/800x800?r=3"
            />
          </div>
          <div
            class="c22"
          >
            <img
              alt="imagem 4"
              src="https://source.unsplash.com/random/800x800?r=4"
            />
          </div>
          <div
            class="c22"
          >
            <img
              alt="imagem 5"
              src="https://source.unsplash.com/random/800x800?r=5"
            />
          </div>
          <div
            class="c22"
          >
            <img
              alt="imagem 6"
              src="https://source.unsplash.com/random/800x800?r=6"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="c11"
      id="teste"
    >
      <div
        class="c3 c4"
      >
        <div
          class="c23"
        >
          <div
            class="c24"
          >
            <h2
              class="c7 c13"
            >
              Grid Two columns
            </h2>
            <p
              class="c15 c16"
            >
              Teste
            </p>
          </div>
          <div
            class="c25"
          >
            <img
              alt="Grid Two columns"
              class="c26"
              src="public/assets/images/javascript.svg"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="c17"
      id="teste"
    >
      <div
        class="c3 c4"
      >
        <div
          class="c18"
        >
          <h2
            class="c7 c19"
          >
            My grid gallery
          </h2>
          <p
            class="c15 c20"
          >
            Aqui vai ter várias imagens e coisas legais e esse texto aqui é para testar a parada da responsividade vem comigo neném
          </p>
        </div>
        <div
          class="c21"
        >
          <div
            class="c22"
          >
            <img
              alt="imagem 1"
              src="https://source.unsplash.com/random/800x800?r=1"
            />
          </div>
          <div
            class="c22"
          >
            <img
              alt="imagem 2"
              src="https://source.unsplash.com/random/800x800?r=2"
            />
          </div>
          <div
            class="c22"
          >
            <img
              alt="imagem 3"
              src="https://source.unsplash.com/random/800x800?r=3"
            />
          </div>
          <div
            class="c22"
          >
            <img
              alt="imagem 4"
              src="https://source.unsplash.com/random/800x800?r=4"
            />
          </div>
          <div
            class="c22"
          >
            <img
              alt="imagem 5"
              src="https://source.unsplash.com/random/800x800?r=5"
            />
          </div>
          <div
            class="c22"
          >
            <img
              alt="imagem 6"
              src="https://source.unsplash.com/random/800x800?r=6"
            />
          </div>
        </div>
      </div>
    </div>
    <footer
      class="c27"
    >
      <div
        class="c3 c4"
      >
        <div
          class="c15 c20"
        >
          <p>
            Bah que loucura
          </p>
        </div>
      </div>
    </footer>
  </div>
  <a
    aria-label="Button go to top"
    class="c28"
    href="#"
    title="Voltar ao topo"
  >
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 0h24v24H0V0z"
        fill="none"
      />
      <path
        d="M4 12l1.41 1.41L11 7.83V20h2V7.83l5.58 5.59L20 12l-8-8-8 8z"
      />
    </svg>
  </a>
</div>
`;
